src/EditorFeatures/VisualBasicTest/RemoveSharedFromModuleMembers/RemoveSharedFromModuleMembersTests.vb

```diff
@@ -0,0 +1,153 @@
+﻿' Licensed to the .NET Foundation under one or more agreements.
+' The .NET Foundation licenses this file to you under the MIT license.
+' See the LICENSE file in the project root for more information.
+
+Imports VerifyVB = Microsoft.CodeAnalysis.Editor.UnitTests.CodeActions.VisualBasicCodeFixVerifier(Of
+    Microsoft.CodeAnalysis.Testing.EmptyDiagnosticAnalyzer,
+    Microsoft.CodeAnalysis.VisualBasic.RemoveSharedFromModuleMembers.VisualBasicRemoveSharedFromModuleMembersCodeFixProvider)
+
+Namespace Microsoft.CodeAnalysis.Editor.VisualBasic.UnitTests.RemoveSharedFromModuleMembers
+    Public Class RemoveSharedFromModuleMembersTests
+        <Fact>
+        Public Async Function TestSharedFieldInModule() As Task
+            Dim source = "
+Public Module M
+    Public {|BC30593:Shared|} x As Integer
```

Comment:
is there a test where 'shared' is the first modifier in the list?

Reply:
@CyrusNajmabadi Yes. The failing one due to incorrect handling of trivia. I will fix that.

Reply:
Try using syntax generator. It should handle this

Reply:
@CyrusNajmabadi I was thinking of copying leading trivia from the token. But let me try syntax generator.

Reply:
@CyrusNajmabadi Using the SyntaxGenerator looks challenging here. The suitable overload to use for field declaration is: `SyntaxGenerator.FieldDeclaration(string name, .....)`. This won't handle if a field declaration has multiple variables (e.g. `Private x, y As String`).

Reply:
After some playing with this, it doesn't sound that we can extract a common helper. We need some special handling here (that's beside trivia).
For example:

```vb
Shared x As String
```

must be converted to:

```vb
Private x As String
```

Just removing the `Shared` keyword will generate a compile error in the absence of other modifiers.

Reply:
I think syntaxgenerator has a helper to get and update modifiers though.

Reply:
@CyrusNajmabadi That simplified it a lot! Thanks.